torch/_inductor/decomposition.py

```diff
@@ -404,6 +405,39 @@ def dequantize_per_tensor_tensor_decomp_impl(
     return (input.to(torch.float32) - zero_point) * scale
 
 
+@register_decomposition([aten.grid_sampler_2d])
+@pw_cast_for_opmath
+def grid_sampler_2d(
+    a: torch.Tensor,
+    grid: torch.Tensor,
+    interpolation_mode: int = 0,
+    padding_mode: int = 0,
+    align_corners: bool = False,
+) -> torch.Tensor:
+
+    # We do not expand the grid (_expand_grid=False) on cpu for performance reasons
+    #
+    if (
+        a.device == torch.device("cpu")
+        and interpolation_mode == 0
+        and a.is_contiguous(memory_format=torch.contiguous_format)
+    ):
+        _expand_grid = False
+    else:
+        _expand_grid = True
+    output = decomp_grid_sampler_2d(
+        a,
+        grid=grid,
+        interpolation_mode=interpolation_mode,
+        padding_mode=padding_mode,
+        align_corners=align_corners,
+        _expand_grid=_expand_grid,
+    )
+    if a.is_contiguous(memory_format=torch.channels_last):
+        output = output.contiguous(memory_format=torch.channels_last)
```

Comment:
Leave a comment about what's the issue with eager.

Also, should this be a "suggest_memory_format" to make it more generic?

Reply:
suggest_memory_format does not always suggest CL even if `a.is_contiguous(memory_format=torch.channels_last) is True`. I would keep it like that.